src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -1681,8 +1681,19 @@ public int get() {
 
         public void endRecovery() throws ElasticsearchException {
             store.decRef();
-            onGoingRecoveries.decrementAndGet();
+            int left = onGoingRecoveries.decrementAndGet();
             assert onGoingRecoveries.get() >= 0 : "ongoingRecoveries must be >= 0 but was: " + onGoingRecoveries.get();
+            if (left == 0) {
+                try {
+                    flush(new Engine.Flush().type(Flush.Type.COMMIT_TRANSLOG));
```

Comment:
For the 1.x and master future-port, how are you going to implement the flush? Are you going to set `force` and `waitIfOngoing`?